File: content/problems/2299-strong-password-checker-ii/solutions/single-pass-category-flags/Solution.java
class Solution {
    public boolean strongPasswordCheckerII(String password){
        if(password.length()<8)return false;
        boolean lower=false,upper=false,digit=false,special=false;
        String specials="!@#$%^&*()-+";
        for(int i=0;i<password.length();i++){
            char ch=password.charAt(i);
            if(i>0&&ch==password.charAt(i-1))return false;
            if(Character.isLowerCase(ch))lower=true;
            else if(Character.isUpperCase(ch))upper=true;
            else if(Character.isDigit(ch))digit=true;
            else if(specials.indexOf(ch)>=0)special=true;
        }
        return lower&&upper&&digit&&special;
    }
}

File: content/problems/2299-strong-password-checker-ii/solutions/single-pass-category-flags/Solution.kt
class Solution {
    fun strongPasswordCheckerII(password:String):Boolean{
        if(password.length<8)return false;
        var lower=false;
        var upper=false;
        var digit=false;
        var special=false;
        val specials="!@#\$%^&*()-+";
        for(i in password.indices){
            val ch=password[i];
            if(i>0&&ch==password[i-1])return false;
            when{
                ch.isLowerCase()->lower=true;
                ch.isUpperCase()->upper=true;
                ch.isDigit()->digit=true;
                ch in specials->special=true
            }
        };
        return lower&&upper&&digit&&special
    }
}
